packages/core/src/registries/designTokenRegistry.ts

```diff
@@ -3,13 +3,47 @@ import { builtInStyles, optionalBuiltInStyles } from '../definitions/styles';
 
 export let designTokensRegistry: DesignTokensDefinition = {};
 
+// This function is used to ensure that the composite values are valid since now those composite values are optional.
+// Therefore only border and in the future text related design tokens are/will be checked in this funciton.
+// Ensuring values for simplie key-value design tokens are not neccessary since they are required via typescript.
+const ensureValidCompositeValues = (designTokenDefinition: DesignTokensDefinition) => {
+  // TODO: add validation logic when text related design tokens are added
+
+  // Border validation
+  if (designTokenDefinition.border) {
+    for (const borderKey in designTokenDefinition.border) {
+      let borderValue = designTokenDefinition.border[borderKey];
+      if (!borderValue.width) {
+        borderValue = {
+          ...borderValue,
+          width: '1px',
+        };
+      }
+      if (!borderValue.style) {
+        borderValue = {
+          ...borderValue,
+          style: 'solid',
+        };
+      }
+      if (!borderValue.color) {
+        borderValue = {
+          ...borderValue,
+          color: '#000000',
+        };
+      }
+      designTokenDefinition.border[borderKey] = borderValue;
```

Comment:
If you want to 🏌️‍♂️ this could be condensed a little bit ⛳ 

```suggestion
      designTokenDefinition.border[borderKey] = {
        ...borderValue,
        width: borderValue.width || '1px',
        style: borderValue.style || 'solid',
        color: borderValue.color || '#000000',
      };
```

Reply:
I think I agree with @primeinteger, though I have no idea on the meaning of those emojies :)

Reply:
https://en.wikipedia.org/wiki/Code_golf

Achieving the same result with less characters - although less characters is not always better, of course. Ideally you find the right balance between simplicity and readability since "code golf" can be taken too far if you're basically writing minified code with single-letter variables and other things like that make the code harder to read

Reply:
Yeah but in this case, what you suggested is way better